### ISSUE TITLE
Wrong callback method registered as hook

### ISSUE THREAD

**Issue by bildschirmartist** [terminal42/contao-changelanguage#218]
Hi,
unfortunately the wrong callback method seems to be registered via the hook annotation in the `NavigationListener` classes. Instead of the `__invoke` method, the `onChangelanguageNavigation` method is registered directly. Only the hook inside the `ArticleNavigationListener` class is correctly registered with the `__invoke`method. Here is an excerpt from my debugging (output of $GLOBALS['TL_HOOKS']['changelanguageNavigation']):
![image](https://user-images.githubusercontent.com/2776658/211534109-a6dcdccf-c6fd-44e5-92bf-979adacf6b05.png)

Note: The blue strikethrough entry is from another extension and can be ignored.

In a separate extension I also use the Hook annotation for invokable services (https://docs.contao.org/dev/framework/hooks/#invokable-services) and here unfortunately also the onChangelanguageNavigation is registered instead of the __invoke method. So here I wrote the hook annotation directly to the method as a workaround:
```php
use Contao\CoreBundle\ServiceAnnotation\Hook;
use Contao\PageModel;
use Terminal42\ChangeLanguage\Event\ChangelanguageNavigationEvent;
use Terminal42\ChangeLanguage\EventListener\Navigation\NewsNavigationListener as Terminal42NewsNavigationListener;
[...]
class NewsNavigationListener extends Terminal42NewsNavigationListener
{
    /**
     * Find record based on languageMain field and parent master archive.
     *
     * @param ChangelanguageNavigationEvent $event
     *
     * @Hook("changelanguageNavigation")
     */
    public function __invoke(ChangelanguageNavigationEvent $event): void
    {
    [...]
    }
[...]
}
```

**Comment by aschempp** [comment]
Can you please check if the latest main branch (f7a7e408be82b78aed1bdd0e7d045b345142c193) fixes the problem?

**Comment by rabauss** [comment]
Attention the behavior is still the same but wanted - it's because the prefered name is `onChangelanguageNavigation` before `__invoke`

**Comment by aschempp** [comment]
Fixed in version 3.4.2

**Comment by bildschirmartist** [comment]
Thank you!
